app/main.py

```diff
@@ -1,13 +1,36 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+
+    return {
+        "line": error["line_number"],
+        "column": error["column_number"],
+        "message": error["text"],
+        "name": error["code"],
+        "source": "flake8"
+    }
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+
+    return {
+        "errors":
+            [
+                format_linter_error(error)
+                for error in errors
+            ],
+        "path": file_path,
+        "status": "failed"
+    }
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+
+    return [
+        {
+            "errors": [],
+            "path": list(linter_report)[0],
+            "status": "passed"
+        }] + [
+            format_single_linter_file(key, value)
+            for key, value in linter_report.items()
+            if value != []
```

Comment:
do it without using that additional check and list merging.

Reply:
I fixed everything